src/library_manager/lib_manager.c

```diff
@@ -311,6 +316,8 @@ struct sof_man_fw_desc *lib_manager_get_library_module_desc(int module_id)
 	struct ext_library *_ext_lib = ext_lib_get();
 	uint8_t *buffptr = (uint8_t *)_ext_lib->desc[lib_id];
 
+	if (!buffptr)
```

Comment:
@pjdobrowolski not sure what "lib_id from IPC would not match with lib_id of module" from the commit message means - does it mean, that it can be a valid ID but for a different library or an invalid ID altogether? If the latter - can it cause beyond array bound access? Or is it verified before? @kv2019i maybe we need a follow-up fix up for this PR?

Reply:
There is no ID validation in FW. If we call module ID which is not available in library we would access to memory with random data.